### PR TITLE
fix(react-infobutton): Popover should close when tabbing out of the surface

### PR DESCRIPTION
**Pull request by sopranopillow**
## Previous Behavior

The popover's surface stays open when focus moves out of it.

## New Behavior

When tabbing out of the surface, the popover closes.

### CODE REVIEW

**Comment by sopranopillow** [comment]
This PR should not be merged until #28613 is merged.

File: change/@fluentui-react-infobutton-e6d4cef7-a090-4513-ac13-c1754d642d63.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "fix: Popover should close when tabbing out of the surface.",
+  "packageName": "@fluentui/react-infobutton",
+  "email": "esteban.230@hotmail.com",
+  "dependentChangeType": "patch"
+}
```
Comment (sopranopillow) [inline change/@fluentui-react-infobutton-e6d4cef7-a090-4513-ac13-c1754d642d63.json:7]: This PR should not be merged until #28613 is merged.

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/Avatar-mount_base.html">633</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/Avatar-mount.html">668</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/Button-mount_base.html">322</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/Button-mount.html">316</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/Field-mount_base.html">1108</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/Field-mount.html">1119</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/FluentProvider-mount_base.html">707</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/FluentProvider-mount.html">701</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/FluentProviderWithTheme-mount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/FluentProviderWithTheme-mount.html">81</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">61</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">65</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">70</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">72</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/InfoButton-mount_base.html">13</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/InfoButton-mount.html">15</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/MakeStyles-mount_base.html">848</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/MakeStyles-mount.html">862</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/Persona-mount_base.html">1748</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/Persona-mount.html">1669</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/SpinButton-mount_base.html">1382</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28913/perf-test-react-components/SpinButton-mount.html">1347</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/768de13c5e30222f64b2cc453bc4b134a6e90712'>768de13c5e30222f64b2cc453bc4b134a6e90712</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: change/@fluentui-react-infobutton-e6d4cef7-a090-4513-ac13-c1754d642d63.json (hunk 1 above)
Comment (sopranopillow) [reply change/@fluentui-react-infobutton-e6d4cef7-a090-4513-ac13-c1754d642d63.json:7]: PR above has been merged

File: packages/react-components/react-infobutton/src/components/InfoButton/useInfoButton.tsx [hunk 2]
```diff
@@ -74,5 +82,43 @@ export const useInfoButton_unstable = (props: InfoButtonProps, ref: React.Ref<HT
   state.popover.open = popoverOpen;
   state.popover.onOpenChange = mergeCallbacks(state.popover.onOpenChange, (e, data) => setPopoverOpen(data.open));
 
+  const closeOnTabBack = (ev: KeyboardEvent) => {
+    ev.stopPropagation();
+    if (ev.currentTarget === ev.target && ev.shiftKey && ev.key === 'Tab') {
+      setPopoverOpen(false);
+    }
+  };
+  const closeOnTabForward = (ev: KeyboardEvent) => {
```
Comment (ling1726) [inline packages/react-components/react-infobutton/src/components/InfoButton/useInfoButton.tsx:91]: I think you'd have an easier time using the `focusout` event listener with a callback ref. That way you'd just need to check that as soon as focus leaves the popover, then it should close:

```ts
const focusOutRef = React.useCallback((el) => {
  if (!el) {
    return;
  }

  el.addEventListener('focusout', (e) => {
    const lostFocus = e.target
    const nextFocused = e.relatedTarget

    if (!elementContains(lostFocus, nextFocused)) {
      /* only run when focus is lost from the popover */
    } 
  })
}, []);
```

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Review by mltejera — APPROVED** [review]
Nicely done sir! Tested and works as I'd expect :-)

**Review by micahgodbolt — APPROVED** [review]

**Pull request merged**